src/coreComponents/linearAlgebra/interfaces/BlasLapackLA.cpp

```diff
@@ -789,4 +789,82 @@ void BlasLapackLA::matrixSVD( arraySlice2d< real64 const, MatrixLayout::ROW_MAJO
   }
 }
 
+void BlasLapackLA::matrixEigenvalues( MatColMajor< real64 const > const & A,
+                                      Vec< std::complex< real64 > > const & lambda )
+{
+  GEOSX_ASSERT_MSG( A.size( 0 ) == A.size( 1 ),
+                    "The matrix A must be square" );
+
+  GEOSX_ASSERT_MSG( A.size( 0 ) == lambda.size(),
+                    "The matrix A and lambda have incompatible sizes" );
+
+  // make a copy of A, since dgeev destroys contents
+  array2d< real64, MatrixLayout::COL_MAJOR_PERM > ACOPY( A.size( 0 ), A.size( 1 ) );
+  for( int i = 0; i < A.size( 0 ); ++i )
+  {
+    for( int j = 0; j < A.size( 1 ); ++j )
+    {
+      ACOPY( i, j ) = A( i, j );
+    }
+  }
```

Comment:
```suggestion
  detail::matrixCopy( A, ACOPY );
```

Reply:
It was a nice copy&paste from `matrixSVD`. I can change both of them.